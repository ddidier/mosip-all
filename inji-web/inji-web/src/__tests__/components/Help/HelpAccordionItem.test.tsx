import React from 'react';
import {fireEvent, screen} from '@testing-library/react';
import {HelpAccordionItem} from "../../../components/Help/HelpAccordionItem";
import { renderWithProvider } from '../../../test-utils/mockUtils';


describe("Testing the Layout of Help Accordion",() => {
    test('Check if the layout is matching with the snapshots', () => {
        const openHandler = jest.fn();
        const{asFragment} = renderWithProvider(<HelpAccordionItem id={1} title={"HelpTitle"} content={["HelpContent"]} open={1} setOpen={openHandler} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Check if current help item is not open, it should not show description', () => {
        const openHandler = jest.fn();
        renderWithProvider(<HelpAccordionItem id={1} title={"HelpTitle"} content={["HelpContent"]} open={0} setOpen={openHandler} />);
        const helpElement = screen.getByTestId("Help-Item-Container");
        expect(helpElement.childElementCount).toBe(1);
    });
    test('Check if current help item is open, it should show description', () => {
        const openHandler = jest.fn();
        renderWithProvider(<HelpAccordionItem id={1} title={"HelpTitle"} content={["HelpContent"]} open={1} setOpen={openHandler} />);
        const helpElement = screen.getByTestId("Help-Item-Container");
        expect(helpElement.childElementCount).toBe(2);
    });

});
describe("Testing the Functionality of Help Accordition Item",()=>{
    test('Check if current help item is open, it should show description', () => {
        const openHandler = jest.fn();
        renderWithProvider(<HelpAccordionItem id={1} title={"HelpTitle"} content={["HelpContent"]} open={0} setOpen={openHandler} />);
        const helpElement = screen.getByTestId("Help-Item-Container");
        expect(helpElement.childElementCount).toBe(1);
        const buttonElement = screen.getByTestId("Help-Item-Title-Button");
        fireEvent.click(buttonElement);
        expect(openHandler).toHaveBeenCalled();
    });
});
