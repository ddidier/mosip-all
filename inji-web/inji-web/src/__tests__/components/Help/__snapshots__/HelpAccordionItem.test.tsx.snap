// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layout of Help Accordion Check if the layout is matching with the snapshots 1`] = `
<DocumentFragment>
  <div
    class="rounded-md mb-2 shadow-sm shadow-iw-shadow"
    data-testid="Help-Item-Container"
  >
    <button
      class="w-full p-5 text-left font-bold hover:bg-iw-helpAccordionHover focus:outline-none"
      data-testid="Help-Item-Title-Button"
    >
      <div
        class="flex flex-row text-iw-title"
        data-testid="Help-Item-Title-Text"
      >
        HelpTitle
        <div
          class="flex items-center ms-auto"
        >
          <svg
            data-testid="Help-Item-UpArrow"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 217.9L383 345c9.4 9.4 24.6 9.4 33.9 0 9.4-9.4 9.3-24.6 0-34L273 167c-9.1-9.1-23.7-9.3-33.1-.7L95 310.9c-4.7 4.7-7 10.9-7 17s2.3 12.3 7 17c9.4 9.4 24.6 9.4 33.9 0l127.1-127z"
            />
          </svg>
        </div>
      </div>
    </button>
    <div
      class="p-5 bg-iw-background border-t-2"
      data-testid="Help-Item-Content-Text"
    >
      <p
        class="text-iw-title"
      >
        HelpContent
      </p>
    </div>
  </div>
</DocumentFragment>
`;
